panic: interface conversion: interface {} is nil, not string

### Community Note

Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform Azure Provider Version
Terraform version : hashicorp/null v3.1.1
Terraform Azure Provider version: hashicorp/azurerm v2.99.0

### Affected Resource(s)
azurerm_cdn_endpoint
Terraform Configuration Files
resource "azurerm_cdn_endpoint" "ce" {
  name                = var.storage_account_name
  profile_name        = var.cdn_profile_name
  location            = "global"
  resource_group_name = var.resource_group_name
  origin_host_header  = data.azurerm_storage_account.sa.primary_web_host
  optimization_type   = "GeneralWebDelivery"
  querystring_caching_behaviour = var.querystring_caching_behaviour
  origin {
    name      = replace(data.azurerm_storage_account.sa.primary_web_host, ".", "-")
    host_name = data.azurerm_storage_account.sa.primary_web_host
  }
  geo_filter  {     
  action = "Block"
  relative_path = "/"
  country_codes = var.blocked_country_codes
  }
}

variable:
variable "blocked_country_codes" {
  default     = [""]
  type        = list
}

### Error
**Error: Plugin did not respond**
│ 
│   with module.default.module.cdn_endpoint_frontend_india.azurerm_cdn_endpoint.ce,
│   on ../../../../modules/azure/cdn-endpoint-spa/main.tf line 8, in resource "azurerm_cdn_endpoint" "ce":
│    8: resource "azurerm_cdn_endpoint" "ce" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│   with module.default.module.cdn_endpoint_frontend_eu.azurerm_cdn_endpoint.ce,
│   on ../../../../modules/azure/cdn-endpoint-spa/main.tf line 8, in resource "azurerm_cdn_endpoint" "ce":
│    8: resource "azurerm_cdn_endpoint" "ce" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
╷
│ Error: Request cancelled
│ 
│   with module.default.module.cdn_endpoint_frontend_panda.azurerm_cdn_endpoint.ce,
│   on ../../../../modules/azure/cdn-endpoint-spa/main.tf line 8, in resource "azurerm_cdn_endpoint" "ce":
│    8: resource "azurerm_cdn_endpoint" "ce" {
│ 
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵

**Stack trace from the terraform-provider-azurerm_v2.99.0_x5 plugin:

panic: interface conversion: interface {} is nil, not string**


This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

### Expected Behavior
Empty string being passed in as a document should be gracefully handled.

### Actual Behavior
Panic.

### Steps to Reproduce
Use azurerm_cdn_endpoint with geo_filter:country_codes
**Pass empty string array "[""]" as country_codes**
Run terraform init
Run terraform plan - **Succeeds**
Run terraform apply - **Fails**

--- a/internal/services/cdn/cdn_endpoint_resource.go
+++ b/internal/services/cdn/cdn_endpoint_resource.go
@@ -514,8 +514,10 @@ func expandCdnEndpointGeoFilters(d *pluginsdk.ResourceData) *[]cdn.GeoFilter {
 		countryCodes := make([]string, 0)
 
 		for _, v := range inputCountryCodes {
-			countryCode := v.(string)
-			countryCodes = append(countryCodes, countryCode)
+			if v != nil {
+				countryCode := v.(string)
+				countryCodes = append(countryCodes, countryCode)
+			}
 		}
 
 		filter := cdn.GeoFilter{

